api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/CalculationCohortDefinitionEvaluator.java

```diff
@@ -70,7 +72,7 @@ protected CalculationResultMap doCalculation(
       if (onDate == null) {
         onDate = (Date) context.getParameterValue("endDate");
         if (onDate == null) {
-          onDate = new Date();
+          onDate = clock.getCurrentDate();
```

Comment:
any future plans for which this clock is gonna be used for?

Reply:
No, I just added it to cover all scenarios. Current date will not be used anymore, we are sticking with `onOrBefore` ([MEPTS-171](https://jembiprojects.jira.com/browse/MEPTS-171)).

Reply:
how about if we then get rid of that entire Clock resource!

Reply:
Sure, I will have to remove the unit test also

Reply:
@kaweesi done